pkg/operator/controller/ingress/deployment.go

```diff
@@ -1123,6 +1123,13 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		)
 	}
 
+	// Add router external certificate environment variable.
+	if routeExternalCertificateEnabled {
+		env = append(env,
+			corev1.EnvVar{Name: "ROUTER_EXTERNAL_CERTIFICATE", Value: "true"},
```

Comment:
The naming in `ROUTER_EXTERNAL_CERTIFICATE` doesn't seem to reflect the meaning passed to the router. It looks like we try to pass the name or some sort of reference to an external certificate. While what we do is we try to enable the handling of the external certificate field.
```suggestion
			corev1.EnvVar{Name: "ROUTER_EXTERNAL_CERTIFICATE_ENABLED", Value: "true"},
```

Reply:
Yeah, it's hard to focus on the right name in the future code during the EP review.  I think it's fine to tweak it in the EP if we find a better name down the implementation line.
I looked at the other variables from the router deployment and found [ROUTER_ENABLE_COMPRESSION](https://github.com/openshift/cluster-ingress-operator/blob/d31546e05cc00021e5fbd441ba6dc6739c1bf274/pkg/operator/controller/ingress/deployment.go#L96) or [ROUTER_DISABLE_HTTP2](https://github.com/openshift/cluster-ingress-operator/blob/d31546e05cc00021e5fbd441ba6dc6739c1bf274/pkg/operator/controller/ingress/deployment.go#L75). I think we should follow the same pattern:
```suggestion
			corev1.EnvVar{Name: "ROUTER_ENABLE_EXTERNAL_CERTIFICATE", Value: "true"},
```

Reply:
Sure. @chiragkyal can you please take a note of this change in the EP? This would also need updating the Router implementation.